ui/index.d.ts

```diff
@@ -1021,16 +1014,37 @@ export declare class LocationListStep<T extends ILocationWizardContext> extends
      * @param wizardContext The context of the wizard
      * @param promptSteps The array of steps to include the LocationListStep to
      */
-    public static addStep<T extends ILocationWizardContext>(wizardContext: IActionContext & Partial<ILocationWizardContext>, promptSteps: AzureWizardPromptStep<T>[]): void;
+    public static addStep<T extends ILocationWizardContext>(wizardContext: IActionContext & ISubscriptionContext & Partial<ILocationWizardContext>, promptSteps: AzureWizardPromptStep<T>[]): void;
 
     /**
      * This will set the wizard context's location (in which case the user will _not_ be prompted for location)
      * For example, if the user selects an existing resource, you might want to use that location as the default for the wizard's other resources
+     * This _will_ set the location even if not all providers support it - in the hopes that a related location can be found during `getLocation`
      * @param wizardContext The context of the wizard
      * @param name The name or display name of the location
      */
     public static setLocation<T extends ILocationWizardContext>(wizardContext: T, name: string): Promise<void>;
 
+    /**
+     * Specify a task that will be used to filter locations
+     * @param wizardContext The context of the wizard
+     * @param task A task evaluating to the locations supported by this provider
+     * @param provider The relevant provider (i.e. 'Microsoft.Web')
+     */
+    public static setLocationSubset<T extends ILocationWizardContext>(wizardContext: T, task: Promise<string[]>, provider: string): void;
+
+    /**
+     * Gets the selected location for this wizard.
+     * @param wizardContext The context of the wizard
+     * @param provider If specified, this will check against that provider's supported locations and attempt to find a "related" location if the selected location is not supported.
+     */
+    public static async getLocation<T extends ILocationWizardContext>(wizardContext: T, provider?: string): Promise<AzExtLocation>;
```

Comment:
async is not supported in d.ts 
Similarly in line: 1057, today the latest azureextensionui (0.42.0) package has build error in extension due to this.
Can you please not include async?

Reply:
According to this, just having a Promise<> return type is enough: https://stackoverflow.com/questions/48448151/how-can-i-add-an-async-function-to-a-type-def-file